parsely/src/main/java/com/parsely/parselyandroid/FlushQueue.kt

```diff
@@ -0,0 +1,56 @@
+package com.parsely.parselyandroid
+
+import com.parsely.parselyandroid.JsonSerializer.toParselyEventsPayload
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+
+internal class FlushQueue(
+    private val flushManager: FlushManager,
+    private val repository: QueueRepository,
+    private val restClient: RestClient,
+    private val scope: CoroutineScope
+) {
+
+    private val mutex = Mutex()
+
+    operator fun invoke(skipSendingEvents: Boolean) {
+        scope.launch {
+            mutex.withLock {
+                val eventsToSend = repository.getStoredQueue()
+
+                if (eventsToSend.isEmpty()) {
+                    flushManager.stop()
+                    return@launch
+                }
+
+                if (skipSendingEvents) {
+                    ParselyTracker.PLog("Debug mode on. Not sending to Parse.ly. Otherwise, would sent ${eventsToSend.size} events")
+                    repository.remove(eventsToSend)
+                    return@launch
```

Comment:
Does this need to stop the `flushManager` as well? I guess it'll get stopped the next time it's flushed. 🤔 

I wonder if we should skip starting the `FlushQueue` all together in a case where we don't want the events to be sent. This is probably beyond the scope of this PR though, so I am just making a note of it here.

Reply:
I removed stopping `flushManager` from this part of code in 784a021. 

> I wonder if we should skip starting the FlushQueue all together in a case where we don't want the events to be sent.

I see your point but the `skipSendingEvents` option is used only for tests (debug mode). As it's like `--dry-run`, I wouldn't opt to remove starting `flushManager` then, as this would cause a significant discrepancy between debug and non-debug modes of SDK. In other words: in debug mode, I believe we want to be as close as possible to non-debug. WDYT?